### ISSUE TITLE
Go with differentiable programming

### ISSUE THREAD

**Issue by luiscastro193** [FluxML/model-zoo#130]
Since AlphaGo is already implemented [here](https://github.com/tejank10/AlphaGo.jl) with Flux, it probably shouldn't be hard to implement a similar experiment with differentiable programming.

Would it make sense? Or is it too complex for differentiable programming?

**Comment by tejank10** [comment]
Good idea. I don't think it would be complex for DP, but we would require some changes in AlphaGo.jl, in order to make it differentiable. `step!` function of Go is not differentiable, since it makes use of `setindex!`.

**Comment by luiscastro193** [comment]
In the meantime, may I ask how well could we expect it to perform? I mean, given enough training (even if such a thing is not realistic), could it compete with AlphaGo Zero?
